pkg/proxy/topology_test.go

```diff
@@ -476,3 +476,75 @@ func TestFilterTopologyEndpoint(t *testing.T) {
 		})
 	}
 }
+
+func TestFilterLocalEndpoint(t *testing.T) {
+	cluster := v1.ServiceInternalTrafficPolicyCluster
+	local := v1.ServiceInternalTrafficPolicyLocal
+
+	testCases := []struct {
+		name                  string
+		internalTrafficPolicy *v1.ServiceInternalTrafficPolicyType
+		endpoints             []Endpoint
+		expected              []Endpoint
```

Comment:
YOu'll want to set the gate in this test and carry a bool per test-case

Reply:
done